src/main/java/seedu/apollo/command/module/DeleteModuleCommand.java

```diff
@@ -56,19 +70,104 @@ public void execute(TaskList taskList, Ui ui, Storage storage, ModuleList module
-            if (toDelete == null) {
-                throw new ModuleNotFoundException();
+            if (args.length == 3) {
+                handleMultiCommand(moduleList, ui, storage);
+            } else {
+                // module code is the only argument
+                String moduleCode = args[0];
+                Module toDelete = moduleList.findModule(moduleCode);
+                if (toDelete == null) {
+                    throw new ModuleNotFoundException();
+                }
+                moduleList.remove(toDelete);
+                ui.printModuleDeleteMessage(moduleCode, moduleList);
             }
-            moduleList.remove(toDelete);
-            ui.printModuleDeleteMessage(moduleCode, moduleList);
+
             storage.updateModule(moduleList, calendar);
+
         } catch (ModuleNotFoundException e) {
-            ui.printUnsuccessfulModuleDelete(moduleCode);
+            ui.printUnsuccessfulModuleDelete(args[0]);
             ui.printTotalModularCredits(moduleList);
         } catch (IOException | InvalidSaveFile e) {
             ui.printErrorForIO();
         }
     }
 
+    private void handleMultiCommand(ModuleList moduleList, Ui ui, Storage storage) throws ModuleNotFoundException {
+        String moduleCode = args[0];
+        String command = args[1];
+        String lessonNumber = args[2];
+
+        try {
+            LessonType lessonType = getCommand(command);
+            Module toDelete = moduleList.findModule(moduleCode);
+
+            if (toDelete == null) {
+                throw new ModuleNotFoundException();
+            }
+
+            this.deleteTimetable(toDelete, lessonType, lessonNumber);
+            ui.printModuleLessonDeleteMessage(moduleCode, lessonType, lessonNumber);
+        } catch (IllegalCommandException e) {
+            ui.printInvalidCommand();
+        } catch (ClassNotFoundException e) {
+            ui.printClassNotAdded();
+        }
+    }
+
+    private void deleteTimetable(Module module, LessonType lessonType, String lessonNumber)
+            throws ClassNotFoundException {
+
+        Boolean isFound = false;
+        ArrayList<Timetable> copyList = new ArrayList<>(module.getModuleTimetable());
+
+        //searches for lesson of specified type and number
+        for (Timetable timetable :copyList) {
+            String classNumber = timetable.getClassnumber();
+            String type = timetable.getLessonType();
+            LessonType lessonType1 = determineLessonType(type);
+
+            if (lessonType1 == lessonType && classNumber.equals(lessonNumber)) {
+                module.getModuleTimetable().remove(timetable);
+                isFound = true;
+            }
+        }
+
+        // if the lesson is not found, throws an exception
+        if (!isFound) {
+            throw new ClassNotFoundException();
+        }
+    }
+
+    private LessonType getCommand(String arg) throws IllegalCommandException {
```

Comment:
Perhaps the name of this method could be getLessonType? not sure if that's what you mean but just a suggestion